src/functional/program_processors/runners/gtfn_cpu.py

```diff
@@ -41,6 +41,7 @@ class GTFNExecutor(ppi.ProgramExecutor):
     builder_factory: compiler.BuildSystemProjectGenerator = compiledb.CompiledbFactory()
 
     name: Optional[str] = None
+    apply_transforms: bool = True
```

Comment:
This we need to talk about. If this is really a regular feature of the backend it should be here. If we only need it internally for testing, I would think about building a no-transforms version of this backend in the tests package.

If this is a customization feature that needs to be conveniently available, that should be reflected in the docstring. It should not appear as if any and all step customization options have to be added to every backend.

Pro separate backend:
- does not affect complexity of the public API (think about how the test matrix grows with every customization option)
- uses the modularity of the workflows to full effect.

Cons:
- if needed by many users often, leads to unneccessary boiler plate
- a certain amount of code duplication (potentially limited to the tests though)
- additional work (copy paste the `__call__` method, change one line, give it a name)

Reply:
The main problem here is that actually the passes should not be part of the backend at all, they should run on any backend. I am not sure what the best strategy for this PR is. To me everything beyond this strategy seems to be out of scope. Not so nice alternative, would be to merge the feature of this PR without test.